test/e2e/utils/test_context.go

```diff
@@ -43,11 +43,12 @@ type KBTestContext struct {
 	Resources  string
 	ImageName  string
 	Kubectl    *Kubectl
+	BinName    string
 }
 
 // TestContext init with a random suffix for test KBTestContext stuff,
 // to avoid conflict when running tests synchronously.
-func TestContext(env ...string) (*KBTestContext, error) {
+func TestContext(binName string, env ...string) (*KBTestContext, error) {
```

Comment:
```suggestion
func NewTestContext(binName string, env ...string) (*KBTestContext, error) {
```

Probably a good idea to generalize `KBTestContext` to `TestContext` too.